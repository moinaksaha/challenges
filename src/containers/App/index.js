// @flow

// Necessary Imports
import React, { Component } from 'react'
import { BrowserRouter as Router, Route } from 'react-router-dom';

// Styled components Imports
// Reference - https://github.com/styled-components/styled-components
import styled, { injectGlobal }  from 'styled-components';

import DonationPage from '../../components/DonationPage';
// import loadable from '@loadable/component';
// const DonationPage = loadable(() => import('../../components/DonationPage'));

injectGlobal`
  body {
    margin: 0;
  }
`
// Styled Components
const MainWrapper = styled.div`
  max-width: 1280px;
  margin: 0px auto;
  font-family: 'Lato', sans-serif;
  min-height: 100vh;
  display: flex;
  background: #ffffff;
`
export class App extends Component {
  render() {
    return (
      <MainWrapper>
        <Router>
          <Route exact path="/" component={DonationPage} />
        </Router>
      </MainWrapper>
    );
  }
};

export default App;


