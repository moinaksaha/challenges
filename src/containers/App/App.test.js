import React from 'react'
import renderer from 'react-test-renderer'
import App from './index.js'

it('App: default', () => {
  const component = renderer.create(<App />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})