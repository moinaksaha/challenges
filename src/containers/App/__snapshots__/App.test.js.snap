// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <App /> Snapshot: Enzyme 1`] = `
<styled.div>
  <BrowserRouter>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
  </BrowserRouter>
</styled.div>
`;
